### PR TITLE
feat: support multiple aqua configs [fixes DXJ-641]

### PR DESCRIPTION
**Pull request by shamsartem**

### CODE REVIEW

File: docs/configs/fluence.md [hunk 1]
```diff
@@ -35,6 +33,42 @@ A map of npm aqua dependency versions
...
+| `output`          | string               | **Yes**  | Relative path to the output directory                                                                                                                                                              |
+| `target`          | string               | **Yes**  | Compilation target Possible values are: `ts`, `js`, `air`.                                                                                                                                         |
+| `constants`       | [object](#constants) | No       | A list of constants to pass to the compiler. Constant name must be uppercase                                                                                                                       |
+| `logLevel`        | string               | No       | Log level for the compiler. Default: info. Must be one of: Must be one of: all, trace, debug, info, warn, error, off Possible values are: `all`, `trace`, `debug`, `info`, `warn`, `error`, `off`. |
```
Comment (akim-bow) [inline docs/configs/fluence.md:56]: duplicate `Must be one of:`

File: src/lib/aqua.ts [hunk 2]
```diff
@@ -202,7 +200,46 @@ export const compileToFiles = async ({
...
+            };
+          },
+        ),
+      )
+    : [
+        {
+          compilationResult: await fn(compileArgs),
+          aquaFilePath: compileArgs.filePath,
+        },
+      ];
```
Comment (akim-bow) [inline src/lib/aqua.ts:235]: I don't like the design choice for `compileDirOrFile` function.
I suggest:
1. Put both imports from `aqua-api` inside `compileDirOrFile`
2. Remove `compileFunctionCall` altogether
3. Make 2 function override signature for `compileDirOrFile` - when it's called with aquaCall and just aqua. 
4. After these modifications function could be removed

Another alternative option i see fit is to make 2 separate functions for compiling aqua and aqua call and put shared logic in some helper function.

The reason for this is that passing concrete function into generic function is unnesessary abstraction which obscure the main logic and also makes code harder to read bcs you need to figure out what's called and when.

Implementation notes
```
type compileDirOrFileReturnType = Promise<
  [
    { compilationResult: CompilationResult; aquaFilePath: string }[],
    { compilationResult: CompilationResult; aquaFilePath: string }[],
  ]
>;
type compileDirOrFileReturnType2 = Promise<
  [
    { compilationResult: CallCompilationResult; aquaFilePath: string }[],
    { compilationResult: CallCompilationResult; aquaFilePath: string }[],
  ]
>;

async function compileDirOrFile(
  compileArgs: CompileFuncCallFromPathArgs,
): compileDirOrFileReturnType2;
async function compileDirOrFile(
  compileArgs: CompileFromPathArgs,
): compileDirOrFileReturnType;
async function compileDirOrFile(
  compileArgs: CompileFromPathArgs | CompileFuncCallFromPathArgs,
) {
  const { compileFromPath, compileAquaCallFromPath } = await import(
    "@fluencelabs/aqua-api"
  );

  const fn =
    "funcCall" in compileArgs
      ? (filePath?: string) => {
          return compileAquaCallFromPath({
            ...compileArgs,
            ...(filePath !== undefined ? { filePath } : {}),
          });
        }
      : (filePath?: string) => {
          return compileFromPath({
            ...compileArgs,
            ...(filePath !== undefined ? { filePath } : {}),
          });
        };

  const isDir = (await stat(compileArgs.filePath)).isDirectory();

  const compilationResultsWithFilePaths = isDir
    ? await Promise.all(
      (await getAquaFilesRecursively(compileArgs.filePath)).map(
        async (aquaFilePath) => {
          return {
            compilationResult: await fn(aquaFilePath),
            aquaFilePath,
          };
        },
      ),
    )
    : [
      {
        compilationResult: await fn(),
        aquaFilePath: compileArgs.filePath,
      },
    ];

  return splitErrorsAndResults(compilationResultsWithFilePaths, (result) => {
    return result.compilationResult.errors.length === 0
      ? { result }
      : { error: result };
  });
}
```

File: src/lib/compileAquaAndWatch.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+    const to =
+      compileArgs.outputPath === undefined || compileArgs.dry
+        ? ""
+        : ` to ${color.yellow(compileArgs.outputPath)}`;
+
+    commandObj.logToStderr(
+      `Successfully compiled${aquaConfigName}${from}${to}`,
+    );
+  }
```
Comment (akim-bow) [inline src/lib/compileAquaAndWatch.ts:159]:
```suggestion
  function compileSuccessLog() {
    const aquaConfigName = name === undefined ? "" : `${color.green(name)}`;

    const from = `from ${color.yellow(compileArgs.filePath)}`;

    const to =
      compileArgs.outputPath === undefined || compileArgs.dry
        ? ""
        : `to ${color.yellow(compileArgs.outputPath)}`;

    commandObj.logToStderr(
      `Successfully compiled ${aquaConfigName} ${from} ${to}`,
    );
  }
```

Optional suggestion.

File: src/lib/paths.ts [hunk 4]
```diff
@@ -141,7 +141,7 @@ const getAquaDir = (cwd?: string): string => {
   return join(getSrcPath(cwd), AQUA_DIR_NAME);
 };
 
-const ensureAquaDir = async (): Promise<string> => {
+export const ensureAquaDir = async (): Promise<string> => {
   return ensureDir(getAquaDir());
 };
```
Comment (akim-bow) [inline src/lib/paths.ts:146]: const -> function ?

**Review by akim-bow — APPROVED** [review]

File: src/lib/aqua.ts (hunk 2 above)
Comment (shamsartem) [reply src/lib/aqua.ts]: We decided to leave this discussion to after MVM. Added TODO

**Pull request merged**